### PR TITLE
feat(testing): make playwright default e2e test runner option

### PR DESCRIPTION
**Pull request by Coly010**
## Current Behavior

Cypress is set as the default e2e test runner

## Expected Behavior

Playwright is set as the default e2e test runner

## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/angular/src/generators/utils/testing.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import type { Tree } from '@nx/devkit';
 import { names, readProjectConfiguration, updateJson } from '@nx/devkit';
 import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
 import { Linter } from '@nx/eslint';
-import { UnitTestRunner } from '../../utils/test-runners';
+import { E2eTestRunner, UnitTestRunner } from '../../utils/test-runners';
```
Comment (leosvelperez) [inline packages/angular/src/generators/utils/testing.ts:5]: Not needed? The import was added, but there's no other change adding the usage.

**Review by leosvelperez — APPROVED** [review]

File: packages/angular/src/generators/utils/testing.ts (hunk 1 above)
Comment (Coly010) [reply packages/angular/src/generators/utils/testing.ts]: Ah yeah, missed leftover. Thanks!

**Review by jaysoo — APPROVED** [review]

**Comment by leosvelperez** [comment]
@Coly010 I guess we should also default to the Playwright choice in the CNW prompt, isn't it?

**Review by AgentEnder — COMMENTED** [review]
Do you think it would be worth adding a migration for current workspaces that already contain a cypress project that sets cypress as the default for the various generators inside `nx.json#generators`?

As is, I could see folks being upset if they have a repo with 5+ e2e apps using cypress, and suddenly this swaps to playwright.

**Comment by Coly010** [comment]
@AgentEnder Those generatorDefaults should be set already when they chose `cypress` as their e2e tool though right?

I'm not sure what the "right" thing to do here would be because I can see people also wanting to actually use playwright but might have legacy cypress projects.

I think it could be a case of having some kind of note in release notes, or wait and see the verdict from people after the change has been made.

**Review by FrozenPandaz — APPROVED** [review]

File: packages/next/src/generators/application/lib/add-e2e.ts [hunk 2]
```diff
@@ -68,6 +68,7 @@ export async function addE2e(host: Tree, options: NormalizedSchema) {
       root: options.e2eProjectRoot,
       sourceRoot: joinPathFragments(options.e2eProjectRoot, 'src'),
       targets: {},
+      tags: [],
```
[the remark this answers is not in the document]
Comment (Coly010) [reply packages/next/src/generators/application/lib/add-e2e.ts:71]: It was failing without it. I can check again

File: e2e/react-module-federation/src/react-module-federation.test.ts [hunk 3]
```diff
@@ -43,7 +43,7 @@ describe('React Module Federation', () => {
         const defaultRemotePort = 4201;
 
         runCLI(
-          `generate @nx/react:host ${shell} --remotes=${remote1},${remote2},${remote3} --style=css --no-interactive --skipFormat --js=${js}`
+          `generate @nx/react:host ${shell} --remotes=${remote1},${remote2},${remote3} --e2eTestRunner=cypress --style=css --no-interactive --skipFormat --js=${js}`
```
[the remark this answers is not in the document]
Comment (Coly010) [reply e2e/react-module-federation/src/react-module-federation.test.ts:46]: They perform specific setup of `.cy.ts` files, it felt wrong to change and not have covered.

[the remark this answers is not in the document]
Comment (Coly010) [reply e2e/react-module-federation/src/react-module-federation.test.ts:46]: There may need to be new e2e tests written specifically for playwright, i'll double check which are missing and address in follow up

**Pull request merged**